flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -718,6 +718,36 @@ void createTemporarySystemFunction(
      */
     void createTemporaryTable(String path, TableDescriptor descriptor);
 
+    /**
+     * Registers the given {@link TableDescriptor} as a temporary catalog table.
+     *
+     * <p>The {@link TableDescriptor descriptor} is converted into a {@link CatalogTable} and stored
+     * in the catalog.
+     *
+     * <p>Temporary objects can shadow permanent ones. If a permanent object in a given path exists,
```

Comment:
I read sentence "Temporary objects can shadow permanent ones"
- I am not sure why I would want to do this ; is this shadowing capability documented?
- I suggest this sentence use `table `instead of` object`?

Reply:
- https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/common/#temporary-vs-permanent-tables
- I agree that `tables` instead of `objects` would be easier to digest in this context. On the other hand, the temporary objects concept seems to exist for multiple resources, like views, and having the same sentence might help people recognize this as being the same pattern. I have no strong opinion here though.